### ISSUE TITLE
Possible race conditions in podman system integration tests

### ISSUE THREAD

**Issue by major** [containers/libpod#3700]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

While running the podman system integration tests, we found that the `/no such/command` test has issues tearing down networks for containers:

```
not ok 11 podman run - basic tests
# (from function `is' in file /usr/share/podman/test/system/helpers.bash, line 289,
#  in test file /usr/share/podman/test/system/030-run.bats, line 30)
#   `is "$output" "$expected_output" "podman run $cmd - output"' failed
# $ podman rm --all --force
# $ podman images --all --format {{.Repository}}:{{.Tag}} {{.ID}}
# docker.io/library/fedora:latest   55dc61afd232
# $ podman run docker.io/library/fedora:latest true
# $ podman run docker.io/library/fedora:latest false
# [ rc=1 (expected) ]
# $ podman run docker.io/library/fedora:latest sh -c exit 32
# [ rc=32 (expected) ]
# $ podman run docker.io/library/fedora:latest echo tpLOfp4uurtZMhlopsIMQvSqBxJXeW
# tpLOfp4uurtZMhlopsIMQvSqBxJXeW
# $ podman run docker.io/library/fedora:latest /no/such/command
# time="2019-08-01T11:11:08-04:00" level=error msg="Error deleting network: running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists
# " 
# time="2019-08-01T11:11:08-04:00" level=error msg="Error while removing pod from CNI network "podman": running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists
# " 
# time="2019-08-01T11:11:08-04:00" level=error msg="unable to cleanup network for container 671a288c67b03346a36282a6a68b0d9168046b4405d76059779fba447b6e648a: "error tearing down CNI namespace configuration for container 671a288c67b03346a36282a6a68b0d9168046b4405d76059779fba447b6e648a: running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists\n"" 
# Error: container_linux.go:345: starting container process caused "exec: \"/no/such/command\": stat /no/such/command: no such file or directory"
# : OCI runtime error
# [ rc=127 (expected) ]
# #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
# #|     FAIL: podman run /no/such/command - output
# #| expected: 'Error: .*: starting container process caused .*exec:.*stat /no/such/command: no such file or directory'
# #|   actual: 'time="2019-08-01T11:11:08-04:00" level=error msg="Error deleting network: running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists'
# #|         > '" '
# #|         > 'time="2019-08-01T11:11:08-04:00" level=error msg="Error while removing pod from CNI network "podman": running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists'
# #|         > '" '
# #|         > 'time="2019-08-01T11:11:08-04:00" level=error msg="unable to cleanup network for container 671a288c67b03346a36282a6a68b0d9168046b4405d76059779fba447b6e648a: "error tearing down CNI namespace configuration for container 671a288c67b03346a36282a6a68b0d9168046b4405d76059779fba447b6e648a: running [/usr/sbin/iptables -t nat -N CNI-b2e1e81e05eb3f27a3925a5b --wait]: exit status 1: iptables v1.8.2 (nf_tables): Chain already exists\n"" '
# #|         > 'Error: container_linux.go:345: starting container process caused "exec: \"/no/such/command\": stat /no/such/command: no such file or directory"'
# #|         > ': OCI runtime error'
# #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# # [teardown]
# $ podman pod rm --all --force
# $ podman rm --all --force
# 6163795ff8c6ed18097f6462f8246a0bae963c12f61be2ab0a3a4c3bdba32a22
# be61a80cf5a18fd19491a0c37c551b8ce5f3b08aee1d053defc28f4f45213707
# 671a288c67b03346a36282a6a68b0d9168046b4405d76059779fba447b6e648a
# e032d3953d3aa28cf8435073b8f2f02a35ebc080b8c2af61fec0919278720865
# f4e687f6d5cb84c44ce99dc1b7551451a4881df85d905e66ed873ba91c6bc8e8
```

**Steps to reproduce the issue:**

1. `bats /usr/share/podman/test/system`

**Describe the results you received:**

The test failed with network teardown issues.

**Describe the results you expected:**

I expected the test to pass.

**Additional information you deem important (e.g. issue happens only occasionally):**

We see this issue more often on aarch64 and ppc64le than we see it on x86_64.

**Output of `podman version`:**

```
podman version 1.4.2
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

This issue happens most often on RHEL 8 on ppc64le/aarch64.

**Comment by mheon** [comment]
@edsantiago PTAL

**Comment by major** [comment]
FWIW I am bringing Fedora 30's `podman-1.4.4-4.fc30` down to RHEL 8 on ppc64le to see if anything changes there.

**Comment by edsantiago** [comment]
Um, looks like the usual iptables thing, no?

**Comment by mheon** [comment]
Oh. Oops. Agree.

I'm going to poke the CNI folks again about an update. I think we have fixed builds going into 8.1, so we really need to get them in elsewhere as well.

**Comment by major** [comment]
Thanks for looking into this! I'll be happy to test whenever something is ready.

**Comment by cevich** [comment]
Note: Realizing a fix for the CNI 'chain already exists' race, requires rolling new VM images with updated packages.  I've got two PRs open which will do this, but are blocked by other problems they uncovered.  I expect that drain to un-clog shortly, and they can move forward next week.  Otherwise if someone else wants to do this in a separate PR, that's fine too.

**Comment by mheon** [comment]
This ought to be fixed now that CNI 0.8.1 is available
